File: TTdsa-java/Exercise/src/com/aacirq/math/N11.java
package com.aacirq.math;

public class N11 {
    public static void main(String[] args) {
        N11 sol = new N11();

        // 2
        System.out.println(sol.NumberOf1(10));
    }

    public int NumberOf1(int n) {
        int res = 0;
        while (n != 0) {
            res++;
            n &= (n - 1);
        }
        return res;
    }
}

File: TTdsa-cpp/Exercise/math/N11.cpp
//
// Created by renxin on 2021/6/18.
//

#include "ttdsa.h"

class Solution {
public:
    int NumberOf1(int n) {
        int res = 0;
        while (n != 0) {
            res++;
            n = n & (n - 1);
        }
        return res;
    }
};

int main() {
    Solution sol;
    // 2
    cout << sol.NumberOf1(10) << endl;
    return 0;
}